hybris.c

```diff
@@ -897,6 +897,83 @@ led_state_hammerhead_set_blink(const led_state_hammerhead_t *self,
 }
 
 /* ------------------------------------------------------------------------- *
+ * bacon sysfs controls for one channel in RGB led
+ * ------------------------------------------------------------------------- */
+
+typedef struct
+{
+  const char *brightness;
+  const char *grpfreq;
+  const char *grppwm;
+  const char *blink; 
+  const char *ledreset; 
+} led_paths_bacon_t;
+
+typedef struct
+{
+  int fd_brightness;
+  int fd_grpfreq;
+  int fd_grppwm;
+  int fd_blink;
+  int fd_ledreset;
+
+  int brightness;
+  int freq;
+  int pwm;
+  int blink;
+  int maxval;
+} led_state_bacon_t;
+
+static void
+led_state_bacon_init(led_state_bacon_t *self)
+{
+  self->fd_brightness = -1;
+  self->fd_grpfreq = -1;
+  self->fd_grppwm = -1;
+  self->fd_blink = -1;
+  self->fd_ledreset = -1;
+
+  self->blink = 0;
+  self->maxval = 255; // load from max_brightness?
+}
+
+static void
+led_state_bacon_close(led_state_bacon_t *self)
+{
+  led_util_close_file(&self->fd_brightness);
+  led_util_close_file(&self->fd_grpfreq);
+  led_util_close_file(&self->fd_grppwm);
+  led_util_close_file(&self->fd_blink);
+  led_util_close_file(&self->fd_ledreset);
+}
+
+static bool
+led_state_bacon_probe(led_state_bacon_t *self,
+                           const led_paths_bacon_t *path)
+{
+  bool res = false;
+
+  led_state_bacon_close(self);
+
+  if( !led_util_open_file(&self->fd_brightness, path->brightness)    ||
```

Comment:
Side note: All of the opens in all of the probes should probably be rearranged so that the least commonly available files are checked first so that we do not end up opening and closing "brightness" file N times during the probing.